trunk/source/CACSL2BoogieTranslator/src/de/uni_freiburg/informatik/ultimate/cdt/translation/implementation/base/chandler/MemoryHandler.java

```diff
@@ -748,6 +747,18 @@ public Pair<RValue, CallStatement> getUltimateMemAllocInitCall(final ILocation a
 		return new Pair<>(addressRValue, ultimateAllocCall);
 	}
 
+	private HeapDataArray determineMemoryArrayForType(final CType type) {
+		if (type instanceof CPointer) {
+			mRequiredMemoryModelFeatures.reportPointerOnHeapRequired();
+			return mMemoryModel.getPointerHeapArray();
+		}
+		if (type instanceof final CPrimitive primitive) {
+			mRequiredMemoryModelFeatures.reportDataOnHeapRequired(primitive.getType());
+			return mMemoryModel.getDataHeapArray(primitive.getType());
+		}
+		throw new AssertionError("There is no memory array for the type " + type);
+	}
+
 	/**
 	 * Generates a call of the read procedure and writes the returned value to a temp variable, returned in the
 	 * expression of the returned ResultExpression. Note that we only read simple types from the heap -- when reading

```

Comment:
If I understand correctly, we are no longer necessarily calling a procedure here. If so, we should update the documentation.